### PR TITLE
fix(mcp): validate guest search parameters

### LINKED ISSUE #511: MCP: validate find_guest requires at least one of name or email
## Problem

`find_guest` requires at least one of `name` or `email` to be supplied, but it is unclear whether this is enforced server-side or only documented. If neither is provided, the behaviour is untested because the auth layer blocks unauthenticated access first.

A request with no search parameters should return a clear `422 Unprocessable Entity` (or equivalent MCP error) rather than an empty result set or an unhandled exception.

## Proposed fix

Add explicit parameter validation in `find_guest` (or its underlying query) that raises a descriptive error when both `name` and `email` are absent. Cover this with a unit test that bypasses the auth dependency so it is not masked in future.

## Context

Discovered during MCP exploration. Auth currently prevents reaching the validation path unauthenticated, so this is latent — but worth a test to guard against regressions if the auth layer ever changes.

### PR DESCRIPTION
**Pull request by tjorim**
## Summary
- reject `find_guest` calls that provide neither `name` nor `email`
- raise a descriptive validation error instead of returning an empty result
- cover the validation path with an auth-independent regression test

## Test plan
- `uv run pytest tests/test_mcp_server.py::TestFindGuest -q`

Closes #511

### CODE REVIEW

File: backend/app/mcp_server.py [hunk 1]
```diff
@@ -402,7 +402,7 @@ async def find_guest(
         role = self._require_volunteer()
 
         if not name and not email:
-            return {"guests": [], "message": "Provide at least one of 'name' or 'email' to search."}
+            raise ValueError("Provide at least one of 'name' or 'email' to search.")
```
[the remark this answers is not in the document]
Comment (tjorim) [reply backend/app/mcp_server.py:407]: Fixed in `7aafe45`. `name` and `email` are stripped before validation, so whitespace-only values now raise the descriptive `ValueError`. I also added a regression test for whitespace-only parameters.

Verified with `uv run pytest tests/test_mcp_server.py::TestFindGuest -q` (`5 passed`).
